client/src/screens/SignUp/SignUp.jsx

```diff
@@ -1,38 +1,157 @@
+      setShowModal(true)
+    } else if (newUser.confirm_password !== newUser.password) {
+      setNewUser(prevState => {
+        return {
+          ...prevState,
+          confirm_password: '',
+          password: '',
+        }
+      });
+      setModalError("Passwords do not match. Please try again.")
+      setShowModal(true)
+    } else if (emailError) {
+      setModalError("An account with this email already exists. Please try another email or Sign in.")
+      setShowModal(true);
+      setNewUser(prevState => {
+        return {
+          ...prevState,
+          email: '',
+          confirm_password: '',
+          password: '',
+        }});
+        setEmailError(null);
+    } else {
+      dispatch(signUpUser(newUser));
+      setShowSuccessModal(true);
+    }
   };
 
+  const handleEmailCheck = async (e) => {
+    const emailReq = {email: e.target.value}
+    const res = await checkEmailAuth(emailReq)
+    if (res) {
+      setEmailError(res)
+    }
+  };
+
+  const handleEditProfile = async () => {
+    const {_id} = await verify();
+    navigate(`/users/${_id}/edit`)
+  }
+
   return (
     <div className="sign-up-screen">
-      <Header headerText={header.text} headerTitle={header.title} />
-      <Form formData={signUpForm} formState={[newUser, setNewUser, handleSignUp]} />
+      {showSuccessModal &&
+      <DoubleActionModal 
+        setShowModal={setShowSuccessModal}
+        bodyText="Success! Do you want to finish setting up your profile or try out the Roulette?"
+        leftText="Finish Profile"
+        leftOnClick={() => handleEditProfile()}
+        rightText="Go to Roulette"
+        rightOnClick={() => navigate("/roulette")}/>}
+      {showModal && 
+        <GenericModal 
+          bodyText={modalError} 
+          buttonText="Ok" 
+          setShowModal={setShowModal}/>  }
+        <h4>Create an account</h4>
+        <form className="form sign-up" onSubmit={handleSignUp}>
+          <div className="input-wrapper">
+            <label htmlFor="first_name">Name</label>
+            <input 
+              id="first_name"
+              name="first_name"
+              onChange={(e) => handleChange(e, "first_name", setNewUser)}
+              type="text"
+              value={newUser["first_name"]}
+              />
+          </div>
+          <div className="input-wrapper">
+            <label htmlFor="last_name">Last Name</label>
+            <input 
+              id="last_name"
+              name="last_name"
+              onChange={(e) => handleChange(e, "last_name", setNewUser)}
+              type="text"
+              value={newUser["last_name"]}
+              />
+          </div>
+          <div className="input-wrapper">
+            <label htmlFor="email">Email Address</label>
+            <input 
+              id="email"
+              name="email"
+              onChange={(e) => handleChange(e, "email", setNewUser)}
+              type="text"
```

Comment:
let's change this type to "email" to prevent users from signing up without valid emails
the HTML form will check if the input has an '@' & throw a pop up if it doesnt